src/App.tsx

```diff
@@ -2,24 +1,44 @@
+import React, { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [visiblePerPage, setVisiblePerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(0);
+  const pages: string[][] = Array.from(
+    { length: Math.ceil(items.length / visiblePerPage) },
+    (_, i) =>
+      items.slice(i * visiblePerPage, i * visiblePerPage + visiblePerPage),
+  );
+  const first = pages[currentPage].at(0);
+  const last = pages[currentPage].at(-1);
+  const from = first ? first.split(' ')[1] : 0;
+  const to = last ? last.split(' ')[1] : 0;
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage + 1} (items {from} - {to} of {items.length})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            defaultValue={'5'}
+            onChange={e => {
+              setVisiblePerPage(Number(e.target.value));
+              setCurrentPage(0);
+            }}
```

Comment:
When changing the number of items per page, you're resetting the current page to 0. This could be confusing to the user if they're on a different page. Consider preserving the current page when changing this setting, or at least informing the user that the page will be reset.